[CircleCI] Extended existing try-catch block

PyvesB opened the pull request:
This pull request should resolve issue #1608.

PyvesB commented:
I believe the CI failure is unrelated to this PR.

RedSparr0w commented:
May be worth adding the [`checkErrorResponse()`](https://github.com/badges/shields/blob/master/server.js#L570-L573) helper function,
As it looks like `data` is undefined so i'm assuming this is happening when the request is returning a `404` or `!=200` status code possibly?

paulmelnikow commented:
By the way, to re-run a test in CI, log in to Circle using your Github account. Look for the Workflow **on-commit** link and click it. From the dropdown, choose **Rerun failed jobs**.

PyvesB commented:
@RedSparr0w I've added the `checkErrorResponse` helper function. Not entirely sure why `data` would be undefined. Status codes should now be handled by the helper, I've added a test where we get a 200 but no response data at all. 😉 

@paulmelnikow thanks for the tip!

paulmelnikow commented on services/circleci/circleci.tester.js:
```diff
@@ -25,7 +25,7 @@ t.create('circle ci (valid, with branch)')
 t.create('circle ci (not found)')
   .get('/project/github/PyvesB/EmptyRepo.json')
-  .expectJSON({name: 'build', value: 'Project not found'});
+  .expectJSON({name: 'build', value: 'not found'});
```
I like the explicit message. You can pass a custom "not found" message to `checkErrorResponse` to preserve the old behavior.

PyvesB replied:
Reverted. I've nevertheless switched to a lower case `P` to be more consistent with what we use elsewhere.

RedSparr0w commented:
>Not entirely sure why data would be undefined.

Actually think i was partially wrong about the status code being the problem with it, Think it may have to do with `{json:true}` in the request, if the data that is returned is not JSON then it probably just returns `undefined`.
_(just a guess, haven't had a chance to test it out yet)_
